### PR TITLE
Add extras methods to the Enumerable module

### PR DESCRIPTION
**Pull request by elmejdki**
- Initial commit
- Configure stickler and rubocop
- Add my_each
- Add my_each_with_index
- Add my_select
- Add my_all?
- Add my_any?
- Add my_none?
- Add my_count
- Add my_map
- Add my_inject
- Update my_map
- create readme file

### CODE REVIEW

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
Project requires changes.

**Comment by pelzolga123** [comment]
## Hi Zakariae!  
### Great job so far  :+1: 
Here are required changes, before this project can be approved:
* my_all:
* [ ] When no block or argument is given should return true when none of the collection members are false or nil:
`[ 10, false, 'a', [ ] ].my_all?` - should return `false`
* my_any?
* [ ] When no block or argument is given should return true if at least one of the collection is not false or nil:
`[ 10, false, 'a', [ ] ].my_any?` - should return `true`
* my_none? 
* [ ] When no block or argument is given should return true only if none of the collection members is true:
`[1, 2 ,3].my_none?` - should return false
* my_map 
* [ ] Should return a new array with the results of running block once for every element in enum:
`[1, 2, 3].my_map { |n| n * 2 }` - should return `[2, 4, 6]` 
* [ ] Should return an Enumerator if no block is given:
`[1, 2 ,3].my_map` - should return `Enumerator`
* my_inject 
* [ ] Should combine all elements of enum by applying a binary operation, specified by a block:
` (5..10).my_inject { |sum, n| sum + n } ` - should return 45
* [ ] When a symbol is specified should combine each element of the collection by applying the symbol as a named method:
`(5..10).my_inject(:*)` - should return 151200

* [ ] Disabling `# rubocop:disable Style/CaseEquality` is alright for this project. Other disabled rules should be deleted from main.rb file.
* [ ] Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: main.rb [hunk 1]
```diff
@@ -1,3 +1,165 @@
...
+  def make_array(input)
+    return input.to_a if input.class.name == 'Range'
 
+    input
+  end
+
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
```
Comment (MauricioRobayo) [inline main.rb:13]: Another option could be `times`: `size.times do |i|`.

**Review by MauricioRobayo — APPROVED** [review]
Nothing left to do, please continue, you are ready for bigger challenges!

Status: __Approved__.

File: main.rb (hunk 1 above)
Comment (elmejdki) [reply main.rb:13]: Nice trick next I will try to use it more thank you

**Pull request merged**